Mathlib/Data/Finset/Lattice.lean

```diff
@@ -547,8 +553,62 @@ theorem inf_sup_distrib_right (s : Finset ι) (f : ι → α) (a : α) :
   @sup_inf_distrib_right αᵒᵈ _ _ _ _ _ _
 #align finset.inf_sup_distrib_right Finset.inf_sup_distrib_right
 
+protected theorem codisjoint_inf_right :
+    Codisjoint a (s.inf f) ↔ ∀ ⦃i⦄, i ∈ s → Codisjoint a (f i) :=
+  @Finset.disjoint_sup_right αᵒᵈ _ _ _ _ _ _
+#align finset.codisjoint_inf_right Finset.codisjoint_inf_right
+
+protected theorem codisjoint_inf_left :
+    Codisjoint (s.inf f) a ↔ ∀ ⦃i⦄, i ∈ s → Codisjoint (f i) a :=
+  @Finset.disjoint_sup_left αᵒᵈ _ _ _ _ _ _
+#align finset.codisjoint_inf_left Finset.codisjoint_inf_left
+
+/- ./././Mathport/Syntax/Translate/Expr.lean:177:8: unsupported: ambiguous notation -/
+theorem inf_sup_inf (s : Finset ι) (t : Finset κ) (f : ι → α) (g : κ → α) :
+    s.inf f ⊔ t.inf g = (s ×ᶠ t).inf fun i => f i.1 ⊔ g i.2 :=
+  @sup_inf_sup αᵒᵈ _ _ _ _ _ _ _ _
+#align finset.inf_sup_inf Finset.inf_sup_inf
+
 end OrderTop
 
+section BoundedOrder
+
+variable [BoundedOrder α] [DecidableEq ι]
+
+--TODO: Extract out the obvious isomorphism `(insert i s).pi t ≃ t i ×ˢ s.pi t` from this proof
+theorem inf_sup {κ : ι → Type _} (s : Finset ι) (t : ∀ i, Finset (κ i)) (f : ∀ i, κ i → α) :
+    (s.inf fun i => (t i).sup (f i)) =
+      (s.pi t).sup fun g => s.attach.inf fun i => f _ <| g _ i.2 := by
+  induction' s using Finset.induction with i s hi ih
+  · simp
+  rw [inf_insert, ih, attach_insert, sup_inf_sup]
+  refine' eq_of_forall_ge_iff fun c => _
+  simp only [Finset.sup_le_iff, mem_product, mem_pi, and_imp, Prod.forall,
+    inf_insert, inf_image]
+  refine'
+    ⟨fun h g hg =>
+      h (g i <| mem_insert_self _ _) (fun j hj => g j <| mem_insert_of_mem hj)
+        (hg _ <| mem_insert_self _ _) fun j hj => hg _ <| mem_insert_of_mem hj,
+      fun h a g ha hg => _⟩
+  -- TODO: This `have` must be named to prevent it being shadowed by the internal `this` in `simpa`
+  have aux : ∀ j : { x // x ∈ s }, ↑j ≠ i := fun j : s => ne_of_mem_of_not_mem j.2 hi
+  -- porting note: `simpa` doesn't support placeholders in proof terms
+  have := h (fun j hj => if hji : j = i then cast (congr_arg κ hji.symm) a
+      else g _ <| mem_of_mem_insert_of_ne hj hji) (fun j hj => ?_)
```

Comment:
So `?_` doesn't work in a `simpa using`?

Reply:
Indeed not